docs/DeveloperGuide.md

```diff
@@ -281,47 +296,269 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+
+
+* 4b. RC4HDB finds multiple residents matching the user input.
+    
+    * 4b1. RC4HDB shows a list of all matching residents.
+      
+      Use case ends.
+      
+
+**Use case: Filtering the list of all residents by specific fields**
+
+**MSS**
+1. User wants to see a list of residents that fall under a certain category.
+2. User enters the filter command followed by an optional specifier to filter for any categories or all of the categories, and lastly the tags to filter for.
+3. RC4HDB applies the filter.
+4. RC4HDB shows the filtered list.
+
+    Use case ends.
+
+**Extension**
+
+* 2a. User enters an invalid specifier i.e. one that is not `/all` or `/any`.
+  
+    * 2a1. RC4HDB shows an error message. 
+      
+      Use case resumes at step 2.
+
+
+* 2b. User enters multiple specifiers i.e. both `/all` and `/any`.
+  
+    * 2b1. RC4HDB shows an error message. 
+      
+      Use case resumes at step 2.
+
+
+* 2c. User enters a category that does not exist.
+  
+    * 2c1. RC4HDB shows an error message.
+      
+      Use case resumes at step 2.
+
+
+* 2d. User enters a value that does not exist in the category.
+  
+    * 2d1. RC4HDB shows an error message. 
+      
+      Use case resumes at step 2.
+
+
+**Use case: Deleting a single resident**
+
+**MSS**
+1. Resident moves out of RC4.
+2. User enters the find command followed by the name of the resident that moved out.
+3. RC4HDB searches for the name of the resident.
+4. RC4HDB displays the details of that resident that is found.
+5. User identifies the index of the resident on the displayed list.
+6. User enters the delete command followed by the index.
+7. RC4HDB removes the corresponding indexed resident from the database.
+8. RC4HDB displays the details of that resident that has been deleted.
+   
+    Use case ends.
+
+**Extension**
+
+* 3a. RC4HDB cannot find any resident matching the user input.
+  
+    * 3a1. RC4HDB shows an error message. 
+      
+      Use case resumes at step 2.
+
+
+* 6a. User enters an invalid index.
+  
+    * 6a1. RC4HDB shows an error message. 
+      
+      Use case resumes at step 5.
+
+
+**Use case: Clearing all data**
+
+**MSS**
+1. User wants to clear all data from the current working file.
+2. User enters the clear command.
+3. RC4HDB replaces the current data file with an empty data file with the same name.
+4. RC4HDB shows a success message.
+   
+    Use case ends.
+
+**Use case: Exiting the application**
+
+**MSS**
+1. User has completed his/her tasks and wants to exit the application.
+2. User enters the exit command.
+3. RC4HDB application closes .
+   
+    Use case ends.
+
+**Extension**
+
+* 2a. User clicks on the exit button.
+  
+    * 2a1. RC4HDB application closes. 
+      
+      Use case ends.
+   
+
+**Use case: Importing data from .csv file**
+
+**MSS**
+1. User has a data file with resident’s information, and wants to view it in RC4HDB.
+2. User enters the import command, followed by the full file path of the CSV file.
```

Comment:
Let's change this step to "User uses the import command to import the CSV file."

Reply:
I've updated all similar edits above. Changed all instances of 'enter' to 'uses', and removed any info regarding the application logic/mechanics.